apps/system/test_softlockup/test_softlockup.c

```diff
@@ -0,0 +1,189 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <unistd.h>
+#include <tinyara/config.h>
+#include <pthread.h>
+#include <tinyara/wqueue.h>
+
+#ifdef CONFIG_SYSTEM_SOFTLOCKUP
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: thread_hogging_function
+ *
+ * Description:
+ *   This functions starts the infinte loop to create the hogging scenario.
+ *
+ * Input parameters:
+ *   void
+ *
+ * Returned Value:
+ *   void
+ *
+ ****************************************************************************/
+
+static void *thread_hogging_function(void *parameter)
+{
+	for (;;) {
+		/* Do Nothing and Hog */
+	}
+	return NULL;
+}
+
+/****************************************************************************
+ * Name: thread_function
+ *
+ * Description:
+ *   This function terminates after a finite period.
+ *
+ * Input parameters:
+ *   void
+ *
+ * Returned Value:
+ *   void
+ *
+ ****************************************************************************/
+
+static void *thread_function(void *parameter)
```

Comment:
Can you change a name meaningful to know what it is?

Reply:
Done.